crates/platform-tags/src/tags.rs

```diff
@@ -538,34 +544,34 @@ mod tests {
             tags,
             @r###"
         [
-            "linux_x86_64",
-            "manylinux_2_5_x86_64",
-            "manylinux_2_6_x86_64",
-            "manylinux_2_7_x86_64",
-            "manylinux_2_8_x86_64",
-            "manylinux_2_9_x86_64",
-            "manylinux_2_10_x86_64",
-            "manylinux_2_11_x86_64",
-            "manylinux_2_12_x86_64",
-            "manylinux_2_13_x86_64",
-            "manylinux_2_14_x86_64",
-            "manylinux_2_15_x86_64",
-            "manylinux_2_16_x86_64",
-            "manylinux_2_17_x86_64",
-            "manylinux_2_18_x86_64",
-            "manylinux_2_19_x86_64",
-            "manylinux_2_20_x86_64",
-            "manylinux_2_21_x86_64",
-            "manylinux_2_22_x86_64",
-            "manylinux_2_23_x86_64",
-            "manylinux_2_24_x86_64",
-            "manylinux_2_25_x86_64",
-            "manylinux_2_26_x86_64",
-            "manylinux_2_27_x86_64",
-            "manylinux_2_28_x86_64",
-            "manylinux2010_x86_64",
-            "manylinux2014_x86_64",
             "manylinux1_x86_64",
+            "manylinux2014_x86_64",
+            "manylinux2010_x86_64",
+            "manylinux_2_28_x86_64",
```

Comment:
Just to make sure I understand this -- does this ordering mean that tags `manylinux1` would be preferred over `manylinux_2_28`?

Note that the "legacy" manylinux tags `manylinux1`, `manylinux2010`, and `manylinux2014` have corresponding glibc versions and should be prioritized like their corresponding "new glibc-style" tags.

https://peps.python.org/pep-0600/#legacy-manylinux-tags

> The existing manylinux tags are redefined as aliases for new-style tags:
>
>    manylinux1_x86_64 is now an alias for manylinux_2_5_x86_64
>    manylinux1_i686 is now an alias for manylinux_2_5_i686
>    manylinux2010_x86_64 is now an alias for manylinux_2_12_x86_64
>    manylinux2010_i686 is now an alias for manylinux_2_12_i686
>    manylinux2014_x86_64 is now an alias for manylinux_2_17_x86_64
>    manylinux2014_i686 is now an alias for manylinux_2_17_i686
>    manylinux2014_aarch64 is now an alias for manylinux_2_17_aarch64
>    manylinux2014_armv7l is now an alias for manylinux_2_17_armv7l
>    manylinux2014_ppc64 is now an alias for manylinux_2_17_ppc64
>    manylinux2014_ppc64le is now an alias for manylinux_2_17_ppc64le
>    manylinux2014_s390x is now an alias for manylinux_2_17_s390x


Concretely, we should expect this priority order:

`manylinux_2_28` > ... > `manylinux_2_17` == `manylinux2014` > `manylinux_2_16` > ... > `manylinux_2_12` == `manylinux2010` > `manylinux_2_11` > ... > `manylinux_2_5` == `manylinux1`

With the aliasing, I _think_ you want to prefer new-style over old style (which resolves any ambiguity in the `==` portions above). Typically wheels are tagged like `pyarrow-15.0.1-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl` and include both new-style and old-style tags if applicable.

Reply:
(Please correct me if I'm reading this wrong, I'm basing this on my recollection of `pip` behavior and a quick skim of https://peps.python.org/pep-0600/#package-installers)

Reply:
Thanks for the details! I can futz with the ordering as you've described and I'll double check that we're matching the expected ordering.